source-container-build/app/source_build.py

```diff
@@ -381,23 +381,45 @@ def build_and_push(
     # push to registry
     fd, digest_file = tempfile.mkstemp()
     os.close(fd)
-    push_cmd = [
-        "skopeo",
-        "copy",
-        "--digestfile",
-        digest_file,
-        f"oci://{image_output_dir}:latest-source",
-        f"docker://{dest_image}",
-    ]
-    log.debug("push source image %r", push_cmd)
-    run(push_cmd, check=True)
+    for dest_image in dest_images:
+        push_cmd = [
+            "skopeo",
+            "copy",
+            "--digestfile",
+            digest_file,
+            f"oci://{image_output_dir}:latest-source",
+            f"docker://{dest_image}",
+        ]
+        log.debug("push source image %r", push_cmd)
+        run(push_cmd, check=True)
     try:
         with open(digest_file, "r") as f:
             build_result["image_digest"] = f.read().strip()
     finally:
         os.unlink(digest_file)
 
 
+def fetch_image_manifest_digest(image: str) -> str:
+    cmd = ["skopeo", "inspect", "--format", "{{.Digest}}", f"docker://{image}"]
+    return run(cmd, check=True, text=True, capture_output=True).stdout.strip()
+
+
+def generate_source_images(image: str) -> list[str]:
+    """Generate source container images from the built binary image
+
+    :param image: str, represent the built image.
+    :return: list of generated source container images.
+    """
+    # For backward-compatibility. It will be removed in near future.
+    deprecated_image = f"{image}.src"
+
+    # in format: sha256:1234567
+    digest = fetch_image_manifest_digest(image)
+    triangulate_image = f"{image.split(':')[0]}:{digest.replace(':', '-')}.src"
```

Comment:
Splitting on `:` is technically wrong, since the registry can include a port number: `image-registry.openshift-image-registry.svc:5000`

Reply:
Updated.